docker/build-tools/build-and-push.sh

```diff
@@ -41,8 +41,10 @@ fi
 ${CONTAINER_CLI} ${CONTAINER_BUILDER}  --target build_tools --build-arg "GOLANG_IMAGE=${GOLANG_IMAGE}" --build-arg "ISTIO_TOOLS_SHA=${SHA}" --build-arg "VERSION=${VERSION}" -t "${HUB}/build-tools:${VERSION}" -t "${HUB}/build-tools:${BRANCH}-latest" .
 # shellcheck disable=SC2086
 ${CONTAINER_CLI} ${CONTAINER_BUILDER}  --build-arg "GOLANG_IMAGE=${GOLANG_IMAGE}" --build-arg "ISTIO_TOOLS_SHA=${SHA}" --build-arg "VERSION=${VERSION}" -t "${HUB}/build-tools-proxy:${VERSION}" -t "${HUB}/build-tools-proxy:${BRANCH}-latest" .
+if [[ "$(uname -m)" == "x86_64" ]]; then
```

Comment:
I think this is good for now since its a clear improvement, but we could be cross-building. We can likely make this check a bit smarter once https://github.com/istio/tools/pull/1596/files is ready by looking at TARGET_PLATFORMS perhaps